packages/utils/src/error-handlers.ts

```diff
@@ -0,0 +1,168 @@
+/**
+ * @module ErrorHandlers
+ * This module is designed to provide utility functions for validating
+ * function parameters. It includes functions that throw type errors if
+ * the parameters do not meet specified criteria, such as being defined,
+ * a number, a string, a function, or an array. This module helps ensure
+ * that functions receive the correct types of inputs, enhancing code
+ * reliability and reducing runtime errors.
+ */
+
+import {
+    SupportedType,
+    isArray,
+    isBigInt,
+    isBigNumberish,
+    isDefined,
+    isFunction,
+    isHexadecimal,
+    isNumber,
+    isObject,
+    isString,
+    isStringifiedBigInt,
+    isSupportedType,
+    isType,
+    isUint8Array
+} from "./type-checks"
+
+/**
+ * It throws a type error if the parameter value has not been defined.
+ * @param parameterValue The parameter value.
+ * @param parameterName The parameter name.
+ */
+export function requireDefined(parameterValue: any, parameterName: string) {
+    if (!isDefined(parameterValue)) {
+        throw new TypeError(`Parameter '${parameterName}' is not defined`)
+    }
+}
+
+/**
+ * It throws a type error if the parameter value is not a number.
+ * @param parameterValue The parameter value.
+ * @param parameterName The parameter name.
+ */
+export function requireNumber(parameterValue: number, parameterName: string) {
+    if (!isNumber(parameterValue)) {
+        throw new TypeError(`Parameter '${parameterName}' is not a number`)
+    }
+}
+
+/**
+ * It throws a type error if the parameter value is not a string.
+ * @param parameterValue The parameter value.
+ * @param parameterName The parameter name.
+ */
+export function requireString(parameterValue: string, parameterName: string) {
+    if (!isString(parameterValue)) {
+        throw new TypeError(`Parameter '${parameterName}' is not a string`)
+    }
+}
+
+/**
+ * It throws a type error if the parameter value is not a function.
+ * @param parameterValue The parameter value.
+ * @param parameterName The parameter name.
+ */
+export function requireFunction(parameterValue: Function, parameterName: string) {
+    if (!isFunction(parameterValue)) {
+        throw new TypeError(`Parameter '${parameterName}' is not a function`)
+    }
+}
+
+/**
+ * It throws a type error if the parameter value is not an array.
+ * @param parameterValue The parameter value.
+ * @param parameterName The parameter name.
+ */
+export function requireArray(parameterValue: any[], parameterName: string) {
+    if (!isArray(parameterValue)) {
+        throw new TypeError(`Parameter '${parameterName}' is not an array`)
+    }
+}
+
+/**
+ * It throws a type error if the parameter value is not a uint8array.
```

Comment:
I thought of using it in lowercase to follow the way the `bigint` type is handled. There is the class `BigInt` and the type `bigint`. Let me know if it makes sense for you.

Reply:
I've noticed that for `bigint`.  For `Uint8array` I've only seen this spelling, which is a bit suprising since I'd have assumed `UInt8Array` with a capital I.  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Uint8Array

JavaScript might be making a distinction between primitive types (lowercase - `bigint`) vs. object types (uppercase - `BigInt`) but I'm not familiar enough to be sure.

I don't really have a strong opinion on this issue.  I'm just pointing out an inconsistency which stuck out to me.

Reply:
> JavaScript might be making a distinction between primitive types (lowercase - `bigint`) vs. object types (uppercase - `BigInt`) but I'm not familiar enough to be sure.
 
There isn't really a primitive type in this case, so it might make sense to change a bit doc and names to reflect that we are actually checking an object type maybe?

I think that makes sense. Let me work on it to wrap this PR up.